### DISCUSSION TITLE
How to disable the automatic attribution to the attribute ?

### DISCUSSION THREAD

**Discussion by vico-gs** [api-platform/core#5695] [Q&A] [answered]
Let's say I have an ORM entity called 'Fruit' with an attribute called 'bananas' which is a oneToMany relation with another ORM entity called Banana.
As I want control on the data returned I decise to create an ApiResource that I want independant from doctrine, let's call it FruitResource, it has an attribute that I call collectionOfBanana which is of type Collection, but despite the name there is no link at all with the banana attribute. As I want to use grpahql, I create a resolver so I can get the data I want. So in the resolver i do:

My graphQlOperation looks like:

```
new Query(
      resolver: FruitResourceItem::class,
      class: Fruit::class,
      read: false,
      deserialize: false
),

```
            
And inside the resolver I got this
```
$fruit = $em->getRespository(Fruit::class)->findOneBy(['id' => 1]);

$fruitResource = new FruitResource();

return $fruitResource;
```

And I don't know why but automatically the attribute collectionOfBanana will return the collection of banana from the original entity.
I have to admit that I don't understand how it's possible ! Does anyone have an idea ? and is it possible to disable the automation for an entity ?

**Comment by soyuka** [comment, marked as answer]
Use a provider not a resolver

**Reply by vico-gs** [reply]
Thanks !
